### ISSUE TITLE
ddiff/diff hangs on nil

### ISSUE THREAD

**Issue by pithyless** [lambdaisland/deep-diff#6]
Hi :) I tracked down a weird hanging issue to `deep-diff/diff`. A minimal repro case:

```
❯ clj -Sdeps '{:deps {lambdaisland/deep-diff {:mvn/version "0.0-25"}}}'
Clojure 1.10.0
user=> (require '[lambdaisland.deep-diff :as ddiff])

user=> user=> (ddiff/diff {:foo :bar} {:foo :bar})
{:foo :bar}

user=> (ddiff/diff {:foo :bar} {nil :wat :foo :bar})
{:foo :bar, #lambdaisland.deep_diff.diff.Insertion{:+ nil} :wat}

user=> (ddiff/diff {nil :wat :foo :bar} {:foo :bar})
;;;; Here we just hang; probably falling into an infinite loop/recur?
```

**Comment by pithyless** [comment]
Turns out this was a big in `clj-diff` and has been fixed as of: `tech.droit/clj-diff {:mvn/version "1.0.1"}`

@plexus - can you bump the dependency and cut a new release?

**Comment by plexus** [comment]
Released in `0.0-29`
